### ISSUE TITLE
BUG: Some template literal expressions being parsed as css...

### ISSUE THREAD

**Issue by mikestopcontinues** [stylelint/postcss-css-in-js#16]
> Clearly describe the bug

There are a few cases where template literal expressions are parsed incorrectly. The first is when used with `calc`. I have this happening in a few files...

<img width="705" alt="Screen Shot 2020-03-29 at 12 32 07 PM" src="https://user-images.githubusercontent.com/150434/77855994-df738600-71b9-11ea-9d1b-40929f23b9a1.png">

But the second case is more interesting. It seems to lose track at the first `{`, thinking it's a style declaration block...

<img width="783" alt="Screen Shot 2020-03-29 at 12 33 07 PM" src="https://user-images.githubusercontent.com/150434/77855996-e00c1c80-71b9-11ea-86ce-00c554f712e0.png">

> Which rule, if any, is the bug related to?

`function-calc-no-invalid`
`property-no-unknown`
`declaration-block-trailing-semicolon`

> What CSS is needed to reproduce the bug?

Shown above.

> What configuration is needed to reproduce the bug?

These are the relevant rules, though I think you see it's got more to do with the parser than the rules themselves...

```json
{
  "rules": {
    "function-calc-no-invalid": true,
    "property-no-unknown": true,
    "declaration-block-trailing-semicolon": "always",
  }
}
```

> Which version of stylelint are you using?

```json
{
    "stylelint": "13.2.1",
    "stylelint-a11y": "1.2.3",
    "stylelint-config-contour": "2.2.0",
    "stylelint-config-recommended": "3.0.0",
    "stylelint-config-styled-components": "0.1.1",
    "stylelint-high-performance-animation": "1.5.0",
    "stylelint-media-use-custom-media": "1.0.0",
    "stylelint-no-unsupported-browser-features": "4.0.0",
    "stylelint-order": "4.0.0",
    "stylelint-processor-styled-components": "1.10.0",
}
```

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI `stylelint --aei **/*.{js,css}` and the `vscode-stylelint` plugin.

> What did you expect to happen?

JS to be correctly parsed as JS. No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

I'm including the full list of errors, as these too are affected by inaccurate parsing...

```shell
components/BlindText.js
 13:3  ✖  Unexpected unknown property "light"   property-no-unknown

components/ContactForm.js
 100:34  ✖  Expected a valid expression   function-calc-no-invalid

components/GlobalStyles.js
 39:36  ✖  Unexpected unknown property     property-no-unknown
           "theme"
 39:40  ✖  Expected a trailing semicolon   declaration-block-trailing-semicolon
 43:38  ✖  Unexpected unknown property     property-no-unknown
           "theme"
 43:42  ✖  Expected a trailing semicolon   declaration-block-trailing-semicolon
 44:31  ✖  Unexpected unknown property     property-no-unknown
           "theme"
 44:35  ✖  Expected a trailing semicolon   declaration-block-trailing-semicolon
 45:31  ✖  Unexpected unknown property     property-no-unknown
           "theme"
 45:35  ✖  Expected a trailing semicolon   declaration-block-trailing-semicolon
 46:25  ✖  Unexpected unknown property     property-no-unknown
           "theme"
 46:29  ✖  Expected a trailing semicolon   declaration-block-trailing-semicolon
 47:37  ✖  Unexpected unknown property     property-no-unknown
           "theme"
 47:41  ✖  Expected a trailing semicolon   declaration-block-trailing-semicolon

components/Region.js
  19:37  ✖  Expected "backColor"  value-keyword-case
            to be "backcolor"
  19:51  ✖  Expected "backColor"  value-keyword-case
            to be "backcolor"
  19:59  ✖  Expected a trailing   declaration-block-trailing-semicolon
            semicolon
  46:41  ✖  Expected whitespace   function-whitespace-after
            after ")"
  46:70  ✖  Expected a trailing   declaration-block-trailing-semicolon
            semicolon
 101:18  ✖  Expected "${fontFam"  function-name-case
            to be "${fontfam"
 101:35  ✖  Unexpected missing    font-family-no-missing-generic-family-keyword
            generic font family
 101:35  ✖  Expected quotes       font-family-name-quotes
            around "}"

components/TalkBubble.js
 84:10  ✖  Expected "${getArrowPosition" to be               function-name-case
           "${getarrowposition"
 85:11  ✖  Expected "${getArrowPosition" to be               function-name-case
           "${getarrowposition"
 86:12  ✖  Expected "${getArrowPosition" to be               function-name-case
           "${getarrowposition"
 87:13  ✖  Expected "${getArrowPosition" to be               function-name-case
           "${getarrowposition"

components/admin/AdminUser.js
 53:27  ✖  Expected a valid expression   function-calc-no-invalid
 60:27  ✖  Expected a valid expression   function-calc-no-invalid

components/app/AppContent.js
 14:36  ✖  Expected a valid expression   function-calc-no-invalid

components/app/AppHeader.js
 124:16  ✖  Expected "${fontFam"  function-name-case
            to be "${fontfam"
 124:33  ✖  Unexpected missing    font-family-no-missing-generic-family-keyword
            generic font family
 124:33  ✖  Expected quotes       font-family-name-quotes
            around "}"
 200:3   ✖  Unexpected unknown    property-no-unknown
            property "light"

components/book/BookCover.js
 70:10  ✖  Expected "${getPageWidth" to be "${getpagewidth"  function-name-case
 71:15  ✖  Expected "${getPageShadow" to be                  function-name-case
           "${getpageshadow"
 75:10  ✖  Expected "${getPageWidth" to be "${getpagewidth"  function-name-case
 76:15  ✖  Expected "${getPageShadow" to be                  function-name-case
           "${getpageshadow"
 80:10  ✖  Expected "${getPageWidth" to be "${getpagewidth"  function-name-case
 81:15  ✖  Expected "${getPageShadow" to be                  function-name-case
           "${getpageshadow"

components/home/HomeBooks.js
 143:33  ✖  Expected a valid expression   function-calc-no-invalid
 147:33  ✖  Expected a valid expression   function-calc-no-invalid
 152:33  ✖  Expected a valid expression   function-calc-no-invalid

components/home/HomeReminder.js
 50:16  ✖  Expected "${fontFam"   function-name-case
           to be "${fontfam"
 50:33  ✖  Unexpected missing     font-family-no-missing-generic-family-keyword
           generic font family
 50:33  ✖  Expected quotes        font-family-name-quotes
           around "}"

components/home/HomeSplash.js
 61:16  ✖  Expected "${fontFam"   function-name-case
           to be "${fontfam"
 61:33  ✖  Unexpected missing     font-family-no-missing-generic-family-keyword
           generic font family
 61:33  ✖  Expected quotes        font-family-name-quotes
           around "}"
```
